src/main/java/com/denizenscript/ddiscordbot/commands/DiscordInteractionCommand.java

```diff
@@ -0,0 +1,564 @@
+    //
+    // @Usage
+    // Use to create a simple slash command without options, which is disabled by default, and save it.
+    // - ~discordinteraction id:mybot command create group:<discord[mybot].group[Denizen]> name:hello "description:Hello world!" enabled:false save:mycmd
+    // - debug log <entry[mycmd].command.name>
+    //
+    // @Usage
+    // Use to create a global slash command with one option.
+    // - ~discordinteraction id:mybot command create name:animal "description:Pick your favorite!" "options:<list_single[<map[type=string;name=animal;description=Your favorite animal]>]>"
+    //
+    // @Usage
+    // Use to edit the permissions of a command.
+    // - ~discordinteraction id:mybot command perms name:mycmd group:<discord[mybot].group[mygroup]> disable_for:<discord[mybot].group[mygroup].role[Muted]> enabled:true
+    //
+    // @Usage
+    // Use to defer and reply to a slash command interaction.
+    // - ~discordinteraction id:mybot defer interaction:<context.interaction>
+    // - ~discordinteraction id:mybot reply interaction:<context.interaction> <context.options.get[hello].if_null[world]>
+    //
+    // @Usage
+    // Use to defer and reply to an interaction ephermally.
+    // - ~discordinteraction id:mybot defer interaction:<context.interaction> ephermal:true
+    // - ~discordinteraction id:mybot reply interaction:<context.interaction> "Shh, don't tell anyone!"
+    //
+    // -->
+
+    public enum DiscordInteractionInstruction { COMMAND, DEFER, REPLY, DELETE }
+    public enum DiscordInteractionCommandInstruction { CREATE, PERMS, DELETE }
+
+    @Override
+    public void parseArgs(ScriptEntry scriptEntry) throws InvalidArgumentsException {
+        for (Argument arg : scriptEntry.getProcessedArgs()) {
+            if (!scriptEntry.hasObject("id")
+                    && arg.matchesPrefix("id")) {
+                scriptEntry.addObject("id", new ElementTag(CoreUtilities.toLowerCase(arg.getValue())));
+            }
+            else if (!scriptEntry.hasObject("instruction")
+                    && arg.matchesEnum(DiscordInteractionInstruction.values())) {
+                scriptEntry.addObject("instruction", arg.asElement());
+            }
+            else if (!scriptEntry.hasObject("command_instruction")
+                    && arg.matchesEnum(DiscordInteractionCommandInstruction.values())) {
+                scriptEntry.addObject("command_instruction", arg.asElement());
+            }
+            else if (!scriptEntry.hasObject("group")
+                    && arg.matchesPrefix("group")
+                    && arg.matchesArgumentType(DiscordGroupTag.class)) {
+                scriptEntry.addObject("group", arg.asType(DiscordGroupTag.class));
+            }
+            else if (!scriptEntry.hasObject("name")
+                    && arg.matchesPrefix("name")) {
+                scriptEntry.addObject("name", new ElementTag(CoreUtilities.toLowerCase(arg.getValue())));
```

Comment:
Should this be forced to lowercase here? You have code later that implies they can be non-lowercase in Discord

Reply:
Can you refer me to the code that implies that?

Reply:
https://github.com/DenizenScript/dDiscordBot/pull/31/files#diff-2c488de091f79585376ca48ec646029a2f929c25e24e28a84963eb693eed655fR209

Reply:
Not sure why I did that. Will have to look into it.

Reply:
It appears that was just a consistency mistake on my part with the matchString variable.